packages/fuels/src/setup-launch-node-server.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-console */
+import type { LaunchNodeOptions, LaunchNodeResult } from '@fuel-ts/account/test-utils';
+import { launchNode } from '@fuel-ts/account/test-utils';
+import http from 'http';
+import type { AddressInfo } from 'net';
+
+process.setMaxListeners(Infinity);
+
+async function parseBody(req: http.IncomingMessage) {
+  return new Promise<string>((resolve, reject) => {
+    const body: Buffer[] = [];
+    req.on('data', (chunk) => {
+      body.push(chunk);
+    });
+    req.on('end', () => {
+      resolve(JSON.parse(body.length === 0 ? '{}' : Buffer.concat(body).toString()));
+    });
+    req.on('error', reject);
+  });
+}
+
+const cleanupFns: Map<string, Awaited<LaunchNodeResult>['cleanup']> = new Map();
+
+function cleanupAllNodes() {
+  cleanupFns.forEach((fn) => fn());
+}
+
+const server = http.createServer(async (req, res) => {
+  res.setHeader('Access-Control-Allow-Origin', '*');
+
+  if (req.url === '/') {
+    const body = (await parseBody(req)) as LaunchNodeOptions;
```

Comment:
I would wrap this in a try/catch as well just because `parseBody` could throw an error / reject the promise.

```suggestion
try {
      const body = await parseBody(req);

      const node = await launchNode({
        port: '0',
        ...body,
        fuelCorePath: 'fuels-core',
      });

      cleanupFns.set(node.url, () => {
        node.cleanup();
        cleanupFns.delete(node.url);
      });

      res.writeHead(200, { 'Content-Type': 'text/plain' });
      res.end(node.url);
    } catch (err) {
      res.writeHead(500, { 'Content-Type': 'text/plain' });
      res.end('Internal Server Error');
      console.error(err);
    }
    ```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2378/commits/7a1ebbfe1a2944f5aebf3de9e75bd6003ba8b497